sos/cleaner/mappings/__init__.py

```diff
@@ -28,11 +29,33 @@ class SoSMap():
     ignore_short_items = False
     match_full_words_only = False
 
-    def __init__(self):
-        self.dataset = {}
+    def __init__(self, workdir='/tmp'):
+        self.dataset = {}  # TODO: mention the child classes can update it as
+        # well as add_sanitised_item_to_dataset method, so the files dont need
+        # to have all counter values stored; e.g. when adding an IP address.
+        # Our approach guarantees same input is added in all processes in the
+        # same ordering - it can add a new network to the dataset.
+        # So dataset bumps by 2
         self._regexes_made = set()
         self.compiled_regexes = []
-        self.lock = Lock()
+        self.cname = self.__class__.__name__.lower()
+        # TODO: ensure the directory <workdir> does exist
+        # TODO: deal with permissions..? IMHO default ones are OK
+        self.workdir = workdir  # the default value is used just by avocado
```

Comment:
Same as above - it may be clearer to add the comment on top (maybe adding a line between the last # TODO and the one in this line)